tests/extract/test_extract_job.py

```diff
@@ -36,18 +42,66 @@ def test_make_job(complex_job, extract_job_settings):
     assert result[0].num == 4
 
 
-# VERIFY SQL OVER DATABASE
 def test_extract_job_sql(extract_job_settings, transactional_db):
+    """
+    VERIFY SQL OVER DATABASE
+    """
     extractor = MySqlSnowflakeExtractor(extract_job_settings.source)
     sql = extractor.get_sql(SQL("SELECT 0"))
     assert "".join(sql.sql.split()) == "".join(EXTRACT_JOB_SQL.split())
 
 
+def test_django_cannot_encode_datetime(extract_job_settings):
+    """
+    DJANGO DOES NOT ENCODE THE DATETIME PROPERLY
+    """
+    epoch = parse(Date.EPOCH).datetime
+    get_ids = SQL(
+        str(
+            (
+                Job.objects.filter(
+                    Q(last_modified__gt=epoch) | (Q(last_modified=epoch) & Q(id__gt=0))
+                )
+                .annotate()
+                .values("id")
+                .order_by("last_modified", "id")[:2000]
+            ).query
+        )
+    )
+    source = MySQL(extract_job_settings.source.database)
+
+    with pytest.raises(Exception):
+        with source.transaction():
+            list(source.query(get_ids, stream=True, row_tuples=True))
```

Comment:
@armenzg It seems the SQL built using `Q` is not properly formed.  I added these tests to demonstrate that.